File: event-bus-rabbit/src/main/java/pegasus/eventbus/topology/events/EventTypeRoutingInfo.java
package pegasus.eventbus.topology.events;

import pegasus.eventbus.amqp.RoutingInfo;

public class EventTypeRoutingInfo {

	final String eventTypeCannonicalName;
	private final RoutingInfo routeInfo;
	
	public EventTypeRoutingInfo(String eventTypeCannonicalName, RoutingInfo routeInfo) {
		super();
		this.eventTypeCannonicalName = eventTypeCannonicalName;
		this.routeInfo = routeInfo;
	}

	public String getEventTypeCannonicalName() {
		return eventTypeCannonicalName;
	}

	public RoutingInfo getRouteInfo() {
		return routeInfo;
	}
}

File: event-bus-rabbit/src/main/dotnet/topology/events/EventTypeRoutingInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using pegasus.eventbus.amqp;


namespace pegasus.eventbus.topology.events
{
    public class EventTypeRoutingInfo
    {
        public string EventTypeCannonicalName;
        public RoutingInfo RouteInfo;


        public EventTypeRoutingInfo() { }
        public EventTypeRoutingInfo(string eventType, RoutingInfo routeInfo)
        {
            this.EventTypeCannonicalName = eventType;
            this.RouteInfo = routeInfo;
        }
    }
}
